File: invoiceJavaBackend/src/main/java/com/example/invoiceJavaBackend/dto/ItemDTO.java
package com.example.invoiceJavaBackend.dto;

import java.math.BigDecimal;

import javax.validation.constraints.NotEmpty;

public class ItemDTO {
    
    @NotEmpty
    private String name;

    private String pkwiu;

    private Integer quantity;

    @NotEmpty
    private String unit;

    private BigDecimal netPrice;

    private BigDecimal netValue;

    private BigDecimal taxRate;

    private BigDecimal taxValue;

    private BigDecimal grossValue;

    public ItemDTO() {
        
    }

    public BigDecimal getGrossValue() {
        return grossValue;
    }

    public void setGrossValue(BigDecimal grossValue) {
        this.grossValue = grossValue;
    }

    public BigDecimal getTaxValue() {
        return taxValue;
    }

    public void setTaxValue(BigDecimal taxValue) {
        this.taxValue = taxValue;
    }

    public BigDecimal getTaxRate() {
        return taxRate;
    }

    public void setTaxRate(BigDecimal taxRate) {
        this.taxRate = taxRate;
    }

    public BigDecimal getNetValue() {
        return netValue;
    }

    public void setNetValue(BigDecimal netValue) {
        this.netValue = netValue;
    }

    public BigDecimal getNetPrice() {
        return netPrice;
    }

    public void setNetPrice(BigDecimal netPrice) {
        this.netPrice = netPrice;
    }

    public String getUnit() {
        return unit;
    }

    public void setUnit(String unit) {
        this.unit = unit;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    public String getPkwiu() {
        return pkwiu;
    }

    public void setPkwiu(String pkwiu) {
        this.pkwiu = pkwiu;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}

File: InvoiceSharpWPFApp/InvoiceSharpWPFApp/Dto/ItemDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InvoiceSharpWPFApp.Dto
{
    public class ItemDTO : INotifyPropertyChanged
    {
        public ItemDTO()
        {
        }

        public string name { get; set; }

        public string pkwiu { get; set; }

        int Quantity;
        public int quantity
        {
            get
            {
                return Quantity;
            }
            set
            {
                Quantity = value;
                netValue = NetPrice * Quantity;
                taxValue = netValue * TaxRate;
                grossValue = netValue + taxValue;
                NotifyPropertyChanged("netValue");
                NotifyPropertyChanged("grossValue");
                NotifyPropertyChanged("taxValue");
            }
        }

        public string unit { get; set; }

        decimal NetPrice;
        public decimal netPrice
        { 
            get
            {
                return NetPrice;
            }
            set
            {
                NetPrice = value;
                netValue = NetPrice * Quantity;
                taxValue = netValue * TaxRate;
                grossValue = netValue + taxValue;
                NotifyPropertyChanged("netValue");
                NotifyPropertyChanged("grossValue");
                NotifyPropertyChanged("taxValue");
            }
        }

        public decimal netValue { get; set; }

        decimal TaxRate;
        public decimal taxRate
        {
            get
            {
                return TaxRate;
            }
            set
            {
                TaxRate = value;
                taxValue = netValue * TaxRate;
                grossValue = netValue + taxValue;
                NotifyPropertyChanged("grossValue");
                NotifyPropertyChanged("taxValue");
            }
        }

        public decimal taxValue { get; set; }

        public decimal grossValue { get; set; }

        public event PropertyChangedEventHandler PropertyChanged;

        public void NotifyPropertyChanged(string propName)
        {
            if (this.PropertyChanged != null)
                this.PropertyChanged(this, new PropertyChangedEventArgs(propName));
        }
    }
}
